webapp/views.py

```diff
@@ -531,9 +531,15 @@ def advantage_view():
                             "daysTillExpiry"
                         ] = date_difference.days
 
-                    contract["renewal"] = make_renewal(
-                        advantage, contract["contractInfo"]
-                    )
+                    try:
+                        contract["renewal"] = make_renewal(
+                            advantage, contract["contractInfo"]
+                        )
+                    except KeyError:
```

Comment:
A key error is a bit accidental here. I guess either make_renewal should catch the KeyError and return a more explicit ValueError or we just have a comment here explaining why make_renewal would fail with a KeyError.

Reply:
@frankban I agree - I think `make_renewal` should just be generally refactored in a followup, to remove direct dict access and use consts for the several string values in it.